pkg/controller/operators/labeller/filters.go

```diff
@@ -49,15 +52,18 @@ func JobFilter(getConfigMap func(namespace, name string) (metav1.Object, error))
 	}
 }
 
+func ServiceAccountFilter(isServiceAccountReferenced func(namespace, name string) bool) func(object metav1.Object) bool {
+	return func(object metav1.Object) bool {
+		return HasOLMOwnerRef(object) || HasOLMLabel(object) || isServiceAccountReferenced(object.GetNamespace(), object.GetName())
+	}
+}
+
```

Comment:
Am I correct in assuming that these filters that are split out are used in special cases where we don't want (or can't) use the `filters` map?

If so, maybe call out the different use cases somehow? I'm a little concerned that a future commit might add something to the `filters` map that ends up being overwritten by some other code that blindly changes the map key to point to another function.

Reply:
Yeah, these are special filters that can't be expressed in a simple way since they require other state inputs. When we're running these filters in the labelling controller, we can answer their other state needs with informers. When we're doing the at-startup check, though, we give them live client calls. I agree that it's easy to break this if you're not careful but I would suggest that we guard against that with careful review to this code in the future rather than some sort of super-clever coding paradigm. I don't know that this needs to get any more complex than it already is.

Reply:
> can't be expressed in a simple way since they require other state inputs.

Ah I see that now.

> I would suggest that we guard against that with careful review to this code in the future rather than some sort of super-clever coding paradigm. I don't know that this needs to get any more complex than it already is.

Totally agree! But I don't think a helper function would add too much complexity:
```go
type filtersMap map[schema.GroupVersionResource]func(metav1.Object) bool

func mustAddFilter(m filtersMap, gvr schema.GroupVersionResource, filter func(metav1.Object) bool) {
	if _, ok := m[gvr]; ok {
 		panic("programmer error: cannot add filter to map: already contains key " + gvr.String())
 	}
 	m[gvr] = filter
}
```

Reply:
That function covers the one case of error, but the other and more insidious one is that you need to start & run every labeller you check for in `Validate()`

Reply:
No super strong feelings here as long as we're confident we can page this knowledge in fairly easily in the future. Maybe just more well-placed comments is enough.